### ISSUE TITLE
problem with virtuals

### ISSUE THREAD

**Issue by bonesoul** [disjunction/node-cache-manager-mongoose#4]
Basically i'm caching data from my mongoose models which has some virtuals with them and model is defined with options;

 {
	toObject: { virtuals: true },
	toJSON: { virtuals: true }
});

The thing is that when i use memcache strore (https://github.com/theogravity/node-cache-manager-memcached-store), the cache data also contains those virtuals but node-cache-manager-mongoose somehow strips them from my data.

any ideas?

**Comment by bonesoul** [comment]
an toJSON() call on data may fix the issue?

**Comment by bonesoul** [comment]
any chances you can review this?

**Comment by disjunction** [comment]
Hi @bonesoul, I acknowledge the problem, yet I had no time to make a fix. I'll probably look into it on weekend. Meanwhile you're welcome to contribute.

**Comment by disjunction** [comment]
I checked how it works with memcached. In my test it did return the same object as I've just set and thus the function members persisted. But if you look into the cached values in memcache - you'll see only a standard JSON representation is saved. And as expected on the next run all functions were gone.

As far as I can see the cache-manager doesn't support caching objects with functions, which is a valid decision. So the behavior is a feature and not a bug. Closing the issue.
